### DISCUSSION TITLE
Single Odoo instance with multiple domains with seperate database per domain.

### DISCUSSION THREAD

**Discussion by ska-ibees** [Tecnativa/doodba#339] [Q&A]
Hi,

During past days I have been understanding doodba. I wanted to thank you for all the hard work done in this regard.

Now, I am planning to migrate my production server and going to adopt doodba based deployment. 

Could you please help to understand as how the same odoo instance could be used for multiple domains with separate databases?

Currently, I am using  [dbfilter_from_header](https://github.com/OCA/server-tools/tree/12.0/dbfilter_from_header) for this purpose. How can the same be achieved in doodba?

**Comment by obayit** [comment]
AFAIK you can add an array of domains to `domains_prod` inside the `.copier-answers.yaml`.

If you are using the copier template answer the `domains_prod` with a value like this:
```
[{hosts: [db1.main.com, db2.main.com]}]
```

**Edit**
Also, I would like to note that, what you mean is probably not multi instance, but multi database.

**Comment by ska-ibees** [comment]
> 
> 
> AFAIK you can add an array of domains to `domains_prod` inside the `.copier-answers.yaml`.
> 
> If you are using the copier template answer the `domains_prod` with a value like this:
> 
> ```
> [{hosts: [db1.main.com, db2.main.com]}]
> ```
> 
> **Edit**
> Also, I would like to note that, what you mean is probably not multi instance, but multi database.
Thanks for pointin that out. I have changed issue title accordingly. 

I am still exploring doodba and trying to test it with Traefik V2. Let's see how it progess. Meanwhile, could you please share an example of single line yaml input (Which is asked during copier questions for  domains_prod and domains_test). I have tried different combinations but couldn't get it working as expected. 
Thanks again!

**Comment by obayit** [comment]
For `domains_prod` just give it `[{hosts: [db1.localhost}]`, replace db1 with the name of the production database.
My advice is to try it with Traefik v1 as described [here](https://github.com/Tecnativa/doodba-copier-template/blob/stable/docs/faq.md#how-to-bootstrap-the-global-inverse-proxy). Because v2 will probably not work so easily.

I too suffered a lot to get the production config running :( but it is definitely worth it.

**Reply by ska-ibees** [reply]
Well, [{hosts: [db1.example.com}] input returns below:

In .copier-answers.yml :

```
domains_prod:
- hosts:
  - db1.example.com
```

in prod.yaml :
```
...
labels:
      doodba.domain.main: "db1.example.com"
      traefik.enable: "true"
      traefik.domain: "db1.example.com"
...
```

Apparently it looks like a subdomain. Whereas, in the [explaination and sample code](https://github.com/Tecnativa/doodba-copier-template/blob/4c1c4b849376ff9efd66c194cac48f4b0584e7b7/copier.yml#L197-L245) db part is not present.

```
domains_prod:
  type: yaml
  multiline: true
  placeholder: |
    # Main domains, with TLS and cert resolved with "letsencrypt" resolver
    - hosts:
        - www.main.com
```


Am I missing something?

**Comment by ska-ibees** [comment]
So, at what point we need to enter actual production domains? And also how can we add additional domains in future? Apparently it looks like things need to be setup during copier questions as manually editing answers file is discouraged. 
BTW, I am able to test Traefik V2 with SSL on a live server (Single EC2 instance). Now exploring ways to deploy production image in Kubernettes (EKS). Lot of new stuff to digest :)

**Reply by yajo** [reply]
> So, at what point we need to enter actual production domains? And also how can we add additional domains in future? Apparently it looks like things need to be setup during copier questions as manually editing answers file is discouraged.

I'd say you answered yourself. :sweat_smile: 

[The `domains_prod` question](https://github.com/Tecnativa/doodba-copier-template/blob/4c1c4b849376ff9efd66c194cac48f4b0584e7b7/copier.yml#L197-L245) has all the needed explanations.

In case you get confused about YAML syntax, just remember that any valid JSON is a valid YAML, so you can write a one-line JSON with your domain definition.

Since you shouldn't edit answers manually, you could do something like:

```bash
dp='
- hosts:
    - db1.example.com
    - db2.example.com
'
copier --data "domains_prod=$dp" --force
```

**Reply by ska-ibees** [reply]
Thanks Yajo `copier --data "domains_prod=$dp" --force` seems promising method.

**Reply by luismata01** [reply]
Hi 
  I don't understand so much  wher apply this code
dp='
- hosts:
    - db1.example.com
    - db2.example.com
'
copier --data "domains_prod=$dp" --force

Can you explain me please

Thank's

**Reply by yajo** [reply]
It's a command to be executed in the root of your subproject.

**Comment by obayit** [comment]
Not sure about all that XD

But moving from hard install to containers is a wild ride for sure.

**Reply by ska-ibees** [reply]
Thanks obayit. Appreciate your help!

**Comment by ska-ibees** [comment]
Well, `[{hosts: [db1.example.com}]` input returns below:

In .copier-answers.yml :

```yaml
domains_prod:
- hosts:
  - db1.example.com

```
in prod.yaml :

```yaml
...
labels:
      doodba.domain.main: "db1.example.com"
      traefik.enable: "true"
      traefik.domain: "db1.example.com"
...

```
Apparently it looks like a subdomain. Whereas, in the explaination and sample code db part is not present.

```yaml
domains_prod:
  type: yaml
  multiline: true
  placeholder: |
    # Main domains, with TLS and cert resolved with "letsencrypt" resolver
    - hosts:
        - www.main.com
```

Am I missing something?

**Reply by yajo** [reply]
AFAICS it's working properly, what did you expect?

**Reply by ska-ibees** [reply]
You are right. With odoo's db_filter and host matching mechanism, It should work in the format of db1.example.com. I wanted to double confirm as example code shows "www.main.com" format.
